### PR TITLE
fix: bugfix for encrypted contacts

### PR DESCRIPTION
**Pull request by beyond009**
## Description
https://mask.atlassian.net/browse/MF-4908
https://mask.atlassian.net/browse/MF-4909
https://mask.atlassian.net/browse/MF-4922
https://mask.atlassian.net/browse/MF-4923

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/background/services/identity/persona/query.ts [hunk 1]
```diff
@@ -62,6 +62,17 @@ export async function queryPersona(id: PersonaIdentifier): Promise<undefined | P
     return result
 }
 
+export async function queryPersonas(): Promise<undefined | PersonaInformation[]> {
+    let result: Promise<PersonaInformation[]> | undefined
+    await createPersonaDBReadonlyAccess(async (t) => {
+        const personas = await queryPersonasDB({}, t)
+        console.log(personas)
```
Comment (guanbinrui) [inline packages/mask/background/services/identity/persona/query.ts]: Remove log.

File: packages/mask/src/extension/popups/hook/useFriends.ts [hunk 2]
```diff
@@ -3,20 +3,27 @@ import {
...
 import { uniqBy } from 'lodash-es'
+import { isProfileIdentifier } from '@masknet/shared'
 
-export type FriendsInformation = ProfileInformation & {
+export type FriendsInformation = Friends & {
     profiles: BindingProof[]
     id: string
 }
 
+type Friends = {
```
Comment (guanbinrui) [inline packages/mask/src/extension/popups/hook/useFriends.ts]: Friend

File: packages/mask/src/extension/popups/hook/useFriends.ts [hunk 3]
```diff
@@ -48,30 +55,51 @@ export function useFriends(): AsyncStateRetry<FriendsInformation[]> {
             1000,
         )
         if (values.length === 0) return EMPTY_LIST
-        const friends = (await Services.Identity.queryProfilesInformation(values.map((x) => x.profile))).filter(
-            (item) => item.linkedPersona !== undefined,
+        const friends: Friends[] = []
+        await Promise.all(
```
Comment (guanbinrui) [inline packages/mask/src/extension/popups/hook/useFriends.ts]: const friends = await Promise.all

File: packages/mask/src/extension/popups/pages/Friends/Search/index.tsx [hunk 4]
```diff
@@ -44,35 +41,42 @@ export const Search = memo<SearchProps>(function Search({ setSearchValue }) {
...
+        <InputBase
+            className={classes.inputRoot}
+            classes={{
+                focused: classes.inputFocused,
+            }}
+            value={value}
+            onKeyUp={(e) => handleKeyPress(e)}
+            onBlur={(e) => {
+                setTimer(
+                    setTimeout(() => {
```
Comment (guanbinrui) [inline packages/mask/src/extension/popups/pages/Friends/Search/index.tsx:54]: To clear the previous timer first.

File: packages/mask/background/services/identity/persona/query.ts [hunk 5]
```diff
@@ -62,6 +62,16 @@ export async function queryPersona(id: PersonaIdentifier): Promise<undefined | P
     return result
 }
 
+export async function queryPersonas(): Promise<undefined | PersonaInformation[]> {
```
Comment (nuanyang233) [inline packages/mask/background/services/identity/persona/query.ts]: Duplicates the `queryOwnedPersonaInformation` function.

File: packages/mask/src/extension/popups/hook/useFriends.ts [hunk 6]
```diff
@@ -48,30 +59,54 @@ export function useFriends(): AsyncStateRetry<FriendsInformation[]> {
...
         if (values.length === 0) return EMPTY_LIST
-        const friends = (await Services.Identity.queryProfilesInformation(values.map((x) => x.profile))).filter(
-            (item) => item.linkedPersona !== undefined,
-        )
+        const friends: Friend[] = (
+            await Promise.all<Promise<Friend | undefined>>(
+                values.map(async (x) => {
+                    if (isProfileIdentifier(x.profile)) {
+                        const res = await Services.Identity.queryProfilesInformation([x.profile])
+                        if (res[0].linkedPersona !== undefined && res[0].linkedPersona !== currentPersona?.identifier)
```
Comment (nuanyang233) [inline packages/mask/src/extension/popups/hook/useFriends.ts]:
```ts
const target = first(res)
if(target.linkedPersona !== undefined &&  target.linkedPersona !== currentPersona?.identifier) {
 //...
}

```

File: packages/mask/src/extension/popups/hook/useFriends.ts [hunk 7]
```diff
@@ -48,30 +59,54 @@ export function useFriends(): AsyncStateRetry<FriendsInformation[]> {
...
+                                avatar: res[0].avatar,
+                            }
+                        return
+                    } else {
+                        if (x.profile !== currentPersona?.identifier) return { persona: x.profile }
+                        return
+                    }
+                }),
+            )
+        ).filter(emptyFilter)
```
Comment (nuanyang233) [inline packages/mask/src/extension/popups/hook/useFriends.ts:80]: https://www.lodashjs.com/docs/lodash.isUndefined#_isundefinedvalue

File: packages/mask/src/extension/popups/hook/useLoadMore.tsx [hunk 8]
```diff
@@ -0,0 +1,21 @@
+import { useState, useEffect, type RefObject } from 'react'
+import { throttle } from 'lodash-es'
+export function useLoadMore(ref: RefObject<HTMLElement>) {
```
Comment (nuanyang233) [inline packages/mask/src/extension/popups/hook/useLoadMore.tsx]: This method has not been called anywhere.

File: packages/mask/src/extension/popups/pages/Friends/Detail/UI.tsx [hunk 9]
```diff
@@ -50,6 +52,25 @@ const useStyles = makeStyles()((theme) => ({
...
+    empty: {
+        position: 'absolute',
+        top: '50%',
+        left: '50%',
+        transform: 'translate(-50%,-50%)',
+        display: 'flex',
+        alignItems: 'center',
+        flexDirection: 'column',
+        gap: 12,
+        color: theme.palette.text.secondary,
```
Comment (nuanyang233) [inline packages/mask/src/extension/popups/pages/Friends/Detail/UI.tsx]:
```suggestion
        color: theme.palette.maskColor.second,
```

Please confirm if the color is correct. The colors we use are all in the `maskColor` in the figma.

File: packages/mask/src/extension/popups/hook/useFriends.ts (hunk 7 above)
Comment (beyond009) [reply packages/mask/src/extension/popups/hook/useFriends.ts:80]: > https://www.lodashjs.com/docs/lodash.isUndefined#_isundefinedvalue

Need a type guard here

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**